---
layout: page
title: R tutorials
permalink: /r-tutorials/
---
The AmphiBac database relies on open-access tools (R, GitHub, and vsearch) to provide access to and useage of the database. While open-source software is the bee's knees, there can often time be a learning curve as to how to use these tools (we've all been there). As such, we have provided step-by-step tools to guide you through the use of the R-package with as much explanation as we can provide. 

---
<ol>
<li>From raw reads to database ready analysis with QIIME2 (for beginners of high-throughput sequencing analysis of 16S rRNA gene data)- <a href="/16s-workflow/"> tutorial</a> </li>
<li>Importing your data into R from common bioinformatics pacakges (e.g. QIIME2, mothur, etc)- <a href="/importing/"> tutorial</a></li>
<li>Comparing your sequences to the AmphiBac Database- <a href="/comparison/"> tutorial</a></li>
<li>Calculate the percentage of inhibitory/enhancing/noninhibitory taxa- <a href="/calculations/"> tutorial</a> </li>
<li>Create a ASV/OTU table of inhibitory taxa- <a href="/tabeling/"> tutorial</a></li>
<li>Antifungal richness calculations- <a href="/richness/"> tutorial</a></li>
<li>Implement filtering to subset database based on metadata (e.g. country, frog species)- <a href="/filtering/"> tutorial</a></li>
<li>Pull the latest AmphiBac database (one is provided on CRAN, but we do update it on GitHub in between CRAN releases...we're good like that)- <a href="/pulling/"> tutorial</a></li>  
</ol>
